### PR TITLE
NMS-14208: upgrade Groovy to latest 2.x

### PR DESCRIPTION
**Pull request by RangerRick**
This PR upgrades our embedded Groovy to the latest 2.x -- 2.5.16 -- which was recently updated in March.

2.5, in general, should be almost entirely source-compatible with the version we'd been using/embedding in OpenNMS.
The biggest changes have been structural (how the Maven project is published, some things being pulled out into "optional" dependencies, etc.)

This patch _should_ provide a version that's functionally equivalent to what we already were using.
At least, smoke tests pass... ¯\\\_(ツ)\_/¯

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14208

### CODE REVIEW

File: features/remote-poller/src/main/groovy/org/opennms/groovy/poller/remote/ScanGui.groovy [hunk 1]
```diff
@@ -57,7 +57,7 @@ import org.springframework.util.Assert
 class ScanGui extends AbstractGui implements ScanReportHandler, PropertyChangeListener, InitializingBean {
     def m_metadataFieldTypes = new TreeSet<MetadataField>()
     def m_locations = new ArrayList<String>()
-    def m_applications = new HashMap<Set<String>>()
+    def m_applications = new HashMap<String, Set<String>>()
```
Comment (jeffgdotorg) [inline features/remote-poller/src/main/groovy/org/opennms/groovy/poller/remote/ScanGui.groovy:60]: Please bump copyright date in this file

**Review by jeffgdotorg — CHANGES_REQUESTED** [review]
Makes sense to me apart from the one missing copyright bump I requested.

**Review by cgorantla — APPROVED** [review]
LGTM!

**Review by christianpape — APPROVED** [review]
LGTM.

**Review by jeffgdotorg — APPROVED** [review]
Looks good, thanks for updating that one copyright date.

**Pull request merged**